### PR TITLE
Call process_set._setup in init() to point to the correct native lib path

### PR DESCRIPTION
**Pull request by Tixxx**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [ ] Did you write any tests to validate this change?  
- [x] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description
In remote trainers of the horovod spark module, if multiple frameworks are imported in the same process, there's a chance that the _basics in horovod.common.process_set is pointing to the wrong shared lib path resulting in horovod not initialized error even after calling hvd.init(). This calls _setup again before fitting the model to make sure the shared lib is correct.

Fixes # (issue).

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

**Review by chongxiaoc — APPROVED** [review]
I think it is non-relevant, but `(test-cpu-gloo-py3_8-tf2_5_1-keras_none-torch1_8_1-mxnet1_7_0_p2-pyspark3_1_2)` also fails on lightning pyspark mnist example.

**Comment by Tixxx** [comment]
> I think it is non-relevant, but `(test-cpu-gloo-py3_8-tf2_5_1-keras_none-torch1_8_1-mxnet1_7_0_p2-pyspark3_1_2)` also fails on lightning pyspark mnist example.

Yea, it's because Ray was upgraded to 1.7 which needed 3.19.1 protobuf, but tf 2.5.1 was compiled against an older version. I think some of the tests are complaining about protobuf mismatch. Let me try to pin the protobuf version to make tf2.5.1 happy, let's see how Ray reacts..

File: horovod/spark/keras/remote.py [hunk 1]
```diff
@@ -227,6 +227,12 @@ def train(serialized_model, train_rows, val_rows, avg_row_size):
                 reader_factory = make_batch_reader
                 is_batch_reader = True
 
+            # Call _setup again in process set module to point shared lib to tensorflow's module
+            # since the lib path might be overwritten in remote trainer.
+            _horovod.common.process_sets._setup(_horovod.tensorflow.mpi_ops._basics)
```
Comment (tgaddair) [inline horovod/spark/keras/remote.py:232]: This is an interesting one. Is this something we can fix a higher level of abstraction, like by calling `hvd.init()` again? I'm wondering if other non-estimator users might run into the same problem (like for `RayExecutor`).

**Comment by maxhgerlach** [comment]
Woah, good catch!

File: horovod/spark/keras/remote.py (hunk 1 above)
Comment (Tixxx) [reply horovod/spark/keras/remote.py:232]: Yea, that's a good suggestion. I think we can hide it in init(), let me test it out.

Comment (Tixxx) [reply horovod/spark/keras/remote.py]: updated to move this logic to init()

**Review by tgaddair — APPROVED** [review]
LGTM! Can land once we fix the Keras tests on master.

**Pull request merged**